libs/screen/sim/image.ts

```diff
@@ -933,6 +933,412 @@ namespace pxsim.ImageMethods {
+        let shearedY = 0;
+
+        const SHEAR = (x: number, y: number) => {
+            shearedX = fxFloor(x + fxMul(y, shearArgs.xShear));
+            shearedY = fxFloor(y + fxMul(shearedX, shearArgs.yShear));
+            shearedX = fxFloor(shearedX + fxMul(shearedY, shearArgs.xShear));
+        }
+
+        dst.makeWritable();
+
+        if (shearArgs.flip) {
+            for (let y = 0; y < shearArgs.scaledHeight; y += FX_ONE) {
+                for (let x = 0; x < shearArgs.scaledWidth; x += FX_ONE) {
+                    let color = getPixel(
+                        src,
+                        fxToInt(fxDiv((shearArgs.scaledWidth - x - 1), shearArgs.sx)),
+                        fxToInt(fxDiv((shearArgs.scaledHeight - y - 1), shearArgs.sy))
+                    );
+
+                    if (!color) continue;
+
+                    SHEAR(x, y);
+                    setPixel(dst, xDst + fxToInt(shearedX - shearArgs.minX), yDst + fxToInt(shearedY - shearArgs.minY), color);
+                }
+            }
+        }
+        else {
+            for (let y = 0; y < shearArgs.scaledHeight; y += FX_ONE) {
+                for (let x = 0; x < shearArgs.scaledWidth; x += FX_ONE) {
+                    let color = getPixel(
+                        src,
+                        fxToInt(fxDiv(x, shearArgs.sx)),
+                        fxToInt(fxDiv(y, shearArgs.sy))
+                    );
+
+                    if (!color) continue;
+
+                    SHEAR(x, y);
+                    setPixel(dst, xDst + fxToInt(shearedX - shearArgs.minX), yDst + fxToInt(shearedY - shearArgs.minY), color);
+                }
+            }
+        }
+    }
+
+    export function _checkOverlapsScaledRotatedImage(dst: RefImage, src: RefImage, args: RefCollection): boolean {
+        const xDst = args.getAt(0) as number;
+        const yDst = args.getAt(1) as number;
+        if (xDst >= dst._width || yDst >= dst._height) {
+            return false;
+        }
+
+        const shearArgs = parseShearArgs(src, args, 2);
+
+        if (
+            shearArgs.sx <= 0 ||
+            shearArgs.sy <= 0 ||
+            xDst + fxToInt(shearArgs.maxX - shearArgs.minX) < 0 ||
+            yDst + fxToInt(shearArgs.maxY - shearArgs.minY) < 0
+        ) {
+            return false;
+        }
+
+        let shearedX = 0;
+        let shearedY = 0;
+
+        const SHEAR = (x: number, y: number) => {
+            shearedX = fxFloor(x + fxMul(y, shearArgs.xShear));
+            shearedY = fxFloor(y + fxMul(shearedX, shearArgs.yShear));
+            shearedX = fxFloor(shearedX + fxMul(shearedY, shearArgs.xShear));
+        }
+
+
+        if (shearArgs.flip) {
+            for (let y = 0; y < shearArgs.scaledHeight; y += FX_ONE) {
+                for (let x = 0; x < shearArgs.scaledWidth; x += FX_ONE) {
+                    let color = getPixel(
+                        src,
+                        fxToInt(fxDiv((shearArgs.scaledWidth - x - 1), shearArgs.sx)),
```

Comment:
Mixing fixed and float here? Should this use FX_ONE instead of the literal `1`?

Reply:
oh yeah, though i fixed all of these when i fixed that in the C++...